server/src/main/java/org/elasticsearch/common/file/NoChangeOccurredException.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.common.file;
+
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Indicates that {@link AbstractFileWatchingService#processFileChanges()} had no effect,
+ * and listeners should not be notified because there's nothing for them to do.
+ */
+public class NoChangeOccurredException extends ExecutionException {
```

Comment:
This is a bit of an odd name, it implies nothing actually went wrong. How about `ClusterStateApplicationFailedException`?

Reply:
Yeah I did not love it. Trouble is, `AbstractFileWatchingService` has no relation to cluster state, and is otherwise completely unaware of it.

I had also considered having `processFileChanges` return a boolean indicating whether listeners should be notified. I loved that even less.